test/typings/MappingProxy.test.ts

```diff
@@ -343,4 +348,110 @@ describe('Mapping proxy', () => {
         );
         expect(() => proxy.getProxiedPath(createPxth(['registeredUser', 'personalData']))).toThrow();
     });
+
+    it('should getValue from nested path', () => {
+        const proxy = new MappingProxy(
+            {
+                location: createPxth(['core', 'values', 'location_from']),
+                cmpId: createPxth(['core', 'cmp_id_from']),
+            },
+            createPxth(['compound'])
+        );
+
+        const values = {
+            core: {
+                cmp_id_from: 5,
+                values: {
+                    location_from: {
+                        id: 24,
+                    },
+                },
+            },
+        };
+
+        const fn = jest.fn(path => {
+            return deepGet(values, path);
+        });
+        const value = proxy.getValue(createPxth(['compound', 'location', 'id']), fn as <U>(path: Pxth<U>) => U);
+        expect(value).toBe(24);
+    });
+
+    it('should setValue to nested path', () => {
+        const proxy = new MappingProxy(
+            {
+                location: createPxth(['core', 'values', 'location_from']),
+                cmpId: createPxth(['core', 'cmp_id_from']),
+            },
+            createPxth(['compound'])
+        );
+
+        const values = {
+            core: {
+                cmp_id_from: 5,
+                values: {
+                    location_from: {
+                        id: 24,
+                    },
+                },
+            },
+        };
+
+        const defaultSetValue = jest.fn((path: Pxth<number>, value: number) => {
+            deepSet(values, path, value);
+        });
+        proxy.setValue(
+            createPxth(['compound', 'location', 'id']),
+            42,
+            defaultSetValue as <U>(path: Pxth<U>, value: U) => void
+        );
+        expect(values.core.values.location_from.id).toBe(42);
```

Comment:
Check the whole "values" object to ensure that "setValue" does not changes any other fields